### PR TITLE
feat(runtime): add gajae-code (gjc) as a native runtime + LLM backend

### PR DESCRIPTION
**Pull request by Yeachan-Heo**
## Summary

Adds **gajae-code (`gjc`)** as a NEW, distinct Ouroboros runtime **and** LLM backend, mirroring the existing `pi` integration pattern **without altering `pi`**, with full bidirectional parity.

> Note: upstream has no `dev` branch, so this draft targets `main`.

### What's included
- **`GjcRuntime`** — per-task `gjc --mode rpc` subprocess adapter with a **strict fail-closed RPC state machine**: `ready` → optional `set_model(provider/modelId)` → `prompt` (ack ≠ completion, id+command-correlated, fails on `success:false` incl. late same-id) → recognized agent events until `agent_end`. Documented passive lifecycle events (`tool_execution_*`, `auto_compaction_*`, `auto_retry_*`, `ttsr_triggered`, `todo_*`) are allowlisted; `workflow_gate`/`host_tool_*`/`host_uri_*`/`extension_ui_request`/unknown frames raise `UnsupportedGjcRpcFrame` and terminate **in all phases incl. pre-ready**; assistant `stopReason="error"` overrides exit 0. stdin is a PIPE closed **before** `wait()` (gjc exits on stdin close). Telemetry: spawn→ready / prompt-ack / wall time. `targeted_resume=False` (no native continuity in v1).
- **`GjcLLMAdapter`** (`llm.backend: gjc`) — same RPC lifecycle with soft `json`/`json_schema` enforcement (inject directive, retry, `Draft202012Validator`, `ProviderError` on exhaustion). Shared `providers/gjc_rpc_protocol.py` keeps both adapters identical.
- **Frontdoor registration** — `gjc` in `auto/run/mcp/init` Typer enums, `LLMConfig.backend` & `OrchestratorConfig.runtime_backend` literals, `gjc_cli_path` + `get_gjc_cli_path` (`OUROBOROS_GJC_CLI_PATH`), `BackendCapability(gjc)`, provider/runtime factories, `config backend gjc`, and `status health`.
- **`setup --runtime gjc`** — writes config, installs a **renderer-generated capability artifact** under `<agent-dir>/rules/` (closing the gap Pi/Goose left open), and installs a **discoverable** Ouroboros-owned `ooo` bridge extension at `<agent-dir>/extensions/ouroboros-ooo-bridge/index.ts` (exact-prefix `ooo`, exit `0`=handled / `78`=continue, recursion guard, `ctx.cwd` dispatch). Agent-dir resolution honors `GJC_CONFIG_DIR` / `GJC_CODING_AGENT_DIR`.
- **Docs** — `docs/runtime-guides/gjc.md` + capability-guide / capability-matrix / agentos runtime-contract / cli-reference updates.

### How it was produced
`deep-interview` (ambiguity 100%→~17%) → `ralplan` consensus (Planner→Architect→Critic, **BLOCK→CLEAR/APPROVE over two rounds**) → gated `ultragoal` execution. The review loop caught and fixed: runtime hang on stdin-close ordering, lifecycle-event rejection, LLM prompt-ack/fail-closed gaps, a non-discoverable bridge path, and a circular-import/layering violation.

### Verification
- **725 targeted tests pass** (`uv run python -m pytest <gjc suite>`), including fake-RPC lifecycle, all unsupported-frame adversarial cases, and fresh-process import smoke in both orders.
- **Real `ouroboros auto --runtime gjc --complete-product` e2e is pending** an authenticated `gjc` in CI; the fake-RPC integration tests cover the full lifecycle.

### Non-goals
No change to the existing `pi` runtime; no `gjc-rpc` Python dependency; no long-lived RPC worker (future ADR); no bidirectional host/gate/UI frame support in v1 (fail-closed).

### CODE REVIEW

**Comment by Yeachan-Heo** [comment]
### ✅ Real end-to-end verification (gjc 0.4.2, live binary)

The previously-pending real e2e has now been run locally against the actual `gjc --mode rpc` binary (Anthropic-backed). All shipped code paths exercised:

1. **`GjcRuntime.execute_task_to_result`** → live gjc used its write tool to create `hello-gjc.txt` (`'hello-gjc\n'`), `returncode=0`, telemetry `spawn_to_ready_ms=1016 / prompt_ack_ms=1044 / task_wall_ms=11932`, result OK.
2. **Factory chain** → `resolve_agent_runtime_backend("gajae-code") -> "gjc"`; `create_agent_runtime(backend="gjc")` -> `GjcRuntime`; real task wrote `factory-proof.txt` (`'OK\n'`), `returncode=0`.
3. **`GjcLLMAdapter.complete`** (`llm.backend: gjc`) → plain completion returned `PONG`; `json_schema` request returned `{"city":"Seoul","population":9700000}` (extracted + `Draft202012Validator`-validated).
4. **`setup --runtime gjc`** install → bridge written to the discoverable `…/agent/extensions/ouroboros-ooo-bridge/index.ts`; capability artifact written to `…/.gjc/agent/rules/ouroboros-skill-capability-guide.md` with content **byte-identical** to `render_backend_skill_capability_guide("gjc")`; agent-dir resolution honors `GJC_CODING_AGENT_DIR` / `GJC_CONFIG_DIR`.

The strict fail-closed RPC state machine, stdin-close-before-wait (exit 0), event normalization, and telemetry all behaved correctly against the real binary — not just fake-RPC. Plus 725 unit tests green.

**Comment by shaun0927** [comment]
@Yeachan-Heo wtf split PRs, make it more agentic.

**Comment by Yeachan-Heo** [comment]
Seriously?

File: src/ouroboros/backends/capabilities.py [hunk 1]
```diff
@@ -265,12 +265,23 @@ def names(self) -> tuple[str, ...]:
...
+        name="gjc",
+        aliases=("gajae-code", "gajae_code"),
+        supports_runtime=True,
+        supports_llm=True,
+        supports_interview_driver=True,
+        switchable_runtime=True,
+        cli_name="gjc",
+        cli_config_key="gjc_cli_path",
+        supports_tool_envelope=False,
+        skill_execution_capabilities=_GENERIC_SKILL_EXECUTION_CAPABILITIES,
```
Comment (Q00) [inline src/ouroboros/backends/capabilities.py:283]: @Yeachan-Heo you can make skill_execution_capabilities more affordable for GJC

**Comment by Yeachan-Heo** [comment]
CI is now green on this branch (all 9 checks pass; bot review APPROVED at \`f41a499\`). Three small follow-up commits, all driven through the gjc harness/session path:

- \`ruff --fix\`: import-sort (I001), 2 unused imports (F401) in \`gjc_llm_adapter\`, quoted annotation (UP037)
- \`pyproject\`: exclude \`src/ouroboros/gjc_bridge/**\` from the wheel \`packages\` sweep so the \`force-include\` is the single source — fixes the duplicate \`gjc_bridge/index.ts\` wheel-archive path that was failing the wheel-packaging tests
- \`ruff format\` on the gjc runtime/adapter/test files (formatting only, no logic change)

On @shaun0927's point: happy to split this into reviewable stacked PRs (e.g. (1) shared rpc protocol + runtime, (2) LLM adapter, (3) frontdoor/config registration, (4) setup bridge + capability artifact, (5) docs) if that's preferred for review. Flagging here before doing the split so we donot churn the green state unnecessarily.

**Comment by Yeachan-Heo** [comment]
Splitting this into a dependency-ordered, individually CI-green stack of 5 PRs (per @shaun0927's "split PRs, make it more agentic"). This monolith stays open until the stack lands; it is superseded by:

- [stack 1/5] gjc foundation — #1379
- [stack 2/5] GjcRuntime — #1380
- [stack 3/5] GjcLLMAdapter — #1381
- [stack 4/5] setup --runtime gjc + ooo bridge (includes the wheel force-include duplicate fix) — #1382
- [stack 5/5] docs — #1383

Each PR fixes the Ruff Lint / Test (3.12–3.14) blockers in its own scope and is green locally (ruff, mypy, full pytest, and `uv build --wheel` for stack 4).

**Comment by Yeachan-Heo** [comment]
### Hardening pass: edge-case suite + live QA + two real source-bug fixes

Adversarial hardening, consensus-reviewed (deep-interview → ralplan: Architect BLOCK→resolved, Critic REJECT→OKAY).

**Source fixes (real bugs the review surfaced):**
1. **Unbounded post-stdin-close `process.wait()`** in BOTH `GjcRuntime` and `GjcLLMAdapter` — a child that emits `agent_end` then ignores stdin-EOF would hang the caller. Now bounded (`asyncio.wait_for` + 5s shutdown budget → force-terminate → `GjcExitError`/`ProviderError`, never success).
2. **Inconsistent id-correlation** — a supported post-ack event with an unrelated `id` is now fail-closed (`GjcProtocolError`) consistently across runtime + adapter via a shared `validate_supported_event_correlation` helper (adapter changed from silent-ignore).
3. **Stream-termination precedence** — nonzero exit before `agent_end` → `GjcExitError` (returncode+stderr); clean EOF before `agent_end` → `GjcProtocolError`; SIGKILL → error with signal/returncode evidence.

**Edge tests** (deterministic fakes + narrow temp-child lifecycle): ready timeout, huge preamble, split/multibyte JSONL, 50MiB cap, unrelated-id, ack-after-agent_end, duplicate agent_end, mid-stream error, EOF precedence, all unsupported frames in all phases, passive flood, unicode/ANSI deltas, set_model ack errors, resume-ignored, EOF-hang/nonzero/SIGKILL/cancellation; adapter retry/extraction/timeout/empty/error; setup agent-dir precedence/idempotency/drift/byte-equality/atomic-write.

**Live QA (opt-in, never breaks CI):** `live_gjc` marker (default `pytest -m "not live_gjc"` hermetic) + `tools/live_gjc_agentos_qa.py` wrapper with an exercised-vs-skipped receipt.

**Verification:** 782 deterministic tests pass; **live suite EXERCISED 5/5 against real gjc 0.4.2**; red-team QA passed with no blockers; `bun check` clean on the gajae-code side. No `pi` change; no `gjc-rpc` dependency.

**Comment by shaun0927** [comment]
Recheck requested for latest HEAD `2512d511e`. The latest commits address the ouroboros-agent blocker by making `config backend gjc` honor `get_gjc_cli_path()`, adding regression coverage for non-PATH GJC installs, and aligning the GJC bridge docs with the installed `extensions/ouroboros-ooo-bridge/index.ts` path. CI did not create a status rollup for the latest fork-head push, so this comment is intended to requeue bot/check evaluation.

**Comment by shaun0927** [comment]
Closing this draft monolith as superseded by the dependency-ordered GJC stack: #1379, #1380, #1381, #1382, and #1383.\n\nRationale:\n- This PR is explicitly draft and its own thread already states that it was split to make the work reviewable and agentic.\n- Keeping the monolith open creates duplicate review state and stale REQUEST_CHANGES noise against work that should be evaluated in the stack PRs.\n- The valid review findings from this draft are being carried forward to the split PRs instead of merging the broad monolith.\n\nNext target is to make the stack PRs individually aligned with the #961 SSOT / AgentOS direction: runtime reality over optimistic surface area, fail-closed subprocess boundaries, and setup/config/docs consistency.
